feature/platform/transceiver/tests/zr_input_output_power_test/zr_input_output_power_test.go

```diff
@@ -0,0 +1,411 @@
+	defer opticalInputPowerStream.Close()
+	transceiverOpticalOutputPowerStream := samplestream.New(t, tc.dut, transceiverOpticalChannelPath.OpticalChannel().OutputPower().State(), 10*time.Second)
+	defer transceiverOpticalOutputPowerStream.Close()
+
+	transceiverOpticalChanelInputPower, _ := opticalInputPowerStream.Next().Val()
+	transceiverOpticalChanelOutputPower, _ := transceiverOpticalOutputPowerStream.Next().Val()
+
+	verifyValidRxInputPower(t, transceiverOpticalChanelInputPower, false, tc.transceiverOpticalChannelName, tc.transceiverName)
+	verifyValidTxOutputPower(t, transceiverOpticalChanelOutputPower, false, tc.transceiverOpticalChannelName, tc.transceiverName)
+
+	t.Logf("Disbaling the interface = %v ", tc.interfaceName)
+	gnmi.Replace(t, tc.dut, interfacePath.Enabled().Config(), *ygot.Bool(false))
+	gnmi.Await(t, tc.dut, interfacePath.Enabled().State(), time.Minute, *ygot.Bool(false))
+
+	t.Logf("Disabled the interface = %v", tc.interfaceName)
+	transceiverOpticalChanelInputPower, _ = opticalInputPowerStream.Nexts(5)[4].Val()
+	transceiverOpticalChanelOutputPower, _ = transceiverOpticalOutputPowerStream.Nexts(5)[4].Val()
+	verifyValidRxInputPower(t, transceiverOpticalChanelInputPower, true, tc.transceiverOpticalChannelName, tc.transceiverName)
+	verifyValidTxOutputPower(t, transceiverOpticalChanelOutputPower, true, tc.transceiverOpticalChannelName, tc.transceiverName)
+
+	t.Logf("Re-enabling the interface = %v ", tc.interfaceName)
+	gnmi.Replace(t, tc.dut, interfacePath.Enabled().Config(), *ygot.Bool(true))
+	gnmi.Await(t, tc.dut, interfacePath.Enabled().State(), time.Minute, *ygot.Bool(true))
+	t.Logf("Re-enabled the interface = %v", tc.interfaceName)
+
+	transceiverOpticalChanelInputPower, _ = opticalInputPowerStream.Nexts(5)[4].Val()
+	transceiverOpticalChanelOutputPower, _ = transceiverOpticalOutputPowerStream.Nexts(5)[4].Val()
+	verifyValidRxInputPower(t, transceiverOpticalChanelInputPower, false, tc.transceiverOpticalChannelName, tc.transceiverName)
+	verifyValidTxOutputPower(t, transceiverOpticalChanelOutputPower, false, tc.transceiverOpticalChannelName, tc.transceiverName)
+
+}
+
+func TestZrInputOutputPower(t *testing.T) {
+	dut1 := ondatra.DUT(t, "dut1")
+	dut2 := ondatra.DUT(t, "dut2")
+
+	transceiver1Port := dut1.Port(t, "port1")
+	transceiver2Port := dut2.Port(t, "port1")
+	transceiver1Name := gnmi.Get(t, dut1, gnmi.OC().Interface(transceiver1Port.Name()).Transceiver().State())
+	transceiver2Name := gnmi.Get(t, dut2, gnmi.OC().Interface(transceiver2Port.Name()).Transceiver().State())
+
+	configureDUT(t, dut1, transceiver1Name)
+	configureDUT(t, dut2, transceiver2Name)
+
+	transceiver1OpticalChannelName := gnmi.Get(t, dut1, gnmi.OC().Component(transceiver1Name).Transceiver().Channel(0).AssociatedOpticalChannel().State())
+	transceiver2OpticalChannelName := gnmi.Get(t, dut2, gnmi.OC().Component(transceiver2Name).Transceiver().Channel(0).AssociatedOpticalChannel().State())
+
+	testCases := []testData{
```

Comment:
testData should either always be referenced as a pointer (in which case use *testData here) or always as a value, in which case the verify methods should accept testData instead of *testData

Reply:
In the function call iam passing  a reference to the objects of the testCases array. I have renamed the variables in the for loop from testData to testDataObj to avoid confusion in the latest commit.

Reply:
Resolved the issue with addition of  min<=avg/instant<=max for Rx and Tx power